### PR TITLE
update: add DataladHCP1200 datagrabber test

### LINKED ISSUE #4: [ENH] HCP Test Data Set on Gin
We need to create a test data set on Gin (https://gin.g-node.org/G-Node/Info/wiki) that is modeled after the real HCP open access datalad dataset (https://github.com/datalad-datasets/human-connectome-project-openaccess), so that we can test our HCP datalad datagrabbers automatically using github.

### PR DESCRIPTION
**Pull request by synchon**
* [x] fix #4
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry to the [latest changes](../docs/changes/latest.inc)

This PR adds test for DataladHCP1200 datagrabber and thus also adds script for test dataset generation on Gin.

### CODE REVIEW

**Comment by fraimondo** [comment]
It needs a bit more of testing.

The HCP has the possibility of getting several tasks and phase encodings. These are also parameters in the constructor, so to restrict the iterator to certain tasks or phase_encodings. So far the test covers only the first element. @synchon: can you test setting the task / phase encoding parameters and that the iterator works accordingly?

**Comment by synchon** [comment]
Ok sure taking a look.

**Comment by fraimondo** [comment]
So the parametrization test is good to test that all the tasks/phase encoding pairs are well formed. The problem is the iterator. If you set phase/tasks in the constructor of the datagrabber, the iterator should restrict to these elements only. Can you make a test for it?

**Comment by synchon** [comment]
Sure.

**Comment by fraimondo** [comment]
testing the bot

**Comment by synchon** [comment]
> So the parametrization test is good to test that all the tasks/phase encoding pairs are well formed. The problem is the iterator. If you set phase/tasks in the constructor of the datagrabber, the iterator should restrict to these elements only. Can you make a test for it?

I think I have addressed the iterator restriction test. Let me know if it is okay.

File: junifer/datagrabber/hcp.py [hunk 1]
```diff
@@ -196,6 +199,7 @@ def __init__(
             datadir=datadir,
             tasks=tasks,
             phase_encodings=phase_encodings,
+            check_file_existence=check_file_existence,
```
Comment (fraimondo) [inline junifer/datagrabber/hcp.py:202]: This does not need to be here. Only in the pattern one.

File: junifer/datagrabber/tests/test_hcp.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        ("WM", "LR"),
+        ("WM", "RL"),
+        ("RELATIONAL", "LR"),
+        ("RELATIONAL", "RL"),
+        ("EMOTION", "LR"),
+        ("EMOTION", "RL"),
+        ("LANGUAGE", "LR"),
+        ("LANGUAGE", "RL"),
+        ("GAMBLING", "LR"),
+        ("GAMBLING", "RL"),
```
Comment (fraimondo) [inline junifer/datagrabber/tests/test_hcp.py:108]: Can you try a couple of tasks too? Also one task, two phase encodings?

```
(["REST1", "REST2"], ["LR", "RL"]),
("REST1", ["LR", "RL"]),
(["REST1", "REST2"], None),
```

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: junifer/datagrabber/hcp.py (hunk 1 above)
Comment (synchon) [reply junifer/datagrabber/hcp.py:202]: Isn't it safe to keep it here as well considering it's the HCP datagrabber?

**Comment by fraimondo** [comment]
Just two small changes, and test for errors if tasks/phase encodings are wrong.

File: junifer/datagrabber/tests/test_hcp.py (hunk 2 above)
Comment (synchon) [reply junifer/datagrabber/tests/test_hcp.py:108]: The cases you commented should be good for the test right? I mean, we just need a few cases.

Comment (fraimondo) [reply junifer/datagrabber/tests/test_hcp.py:108]: exactly

Comment (synchon) [reply junifer/datagrabber/tests/test_hcp.py:108]: Done.

File: junifer/datagrabber/hcp.py (hunk 1 above)
Comment (synchon) [reply junifer/datagrabber/hcp.py:202]: Or maybe we explicitly pass `False` here?

Comment (synchon) [reply junifer/datagrabber/hcp.py:202]: Okay on second thought, I will remove it, doesn't really harm.

Comment (synchon) [reply junifer/datagrabber/hcp.py]: Done.

File: junifer/datagrabber/hcp.py [hunk 3]
```diff
@@ -199,3 +199,8 @@ def __init__(
             uri=uri,
             rootdir=rootdir,
         )
+
+    @property
```
Comment (fraimondo) [inline junifer/datagrabber/hcp.py:203]: Do we need to have this here? I thought the pattern one would have the default (`False`)

Comment (fraimondo) [reply junifer/datagrabber/hcp.py:203]: Ah, sorry, this is the Datalad one, this should return `True` then.

**Review by fraimondo — CHANGES_REQUESTED** [review]

File: junifer/datagrabber/hcp.py (hunk 3 above)
Comment (synchon) [reply junifer/datagrabber/hcp.py:203]: I believe this is ok then.

**Pull request merged**